### PR TITLE
add callback to expense income budget models to downcase category. Ad…

### PR DESCRIPTION
**Pull request by josephlee702**
…just tests to reflect this

### Keypoints:
- Added a callback to the income, expense, and budget models to downcase them in the JSON response.
- Adjusted tests to reflect these changes.

### Related Issue(s)
List any related issue(s) or task(s) this PR addresses.<br>
Closes DF-74

### Checklist:
- [x] New/Modified Tests are Passing
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings or errors.

### CODE REVIEW

**Comment by josephlee702** [comment]
@trpubz make sure this is what we wanted! This is a before -> after.
image
image

File: app/models/budget.rb [hunk 1]
```diff
@@ -17,6 +15,15 @@ def pct_remaining
     end
   end
 
+  def downcase_category
```
Comment (shawndcarpenter) [inline app/models/budget.rb:20]: yep, this looks like what we were talking about yesterday. all categories should be downcased before being saved

File: spec/graphql/mutations/incomes/create_income_request_spec.rb [hunk 2]
```diff
@@ -30,7 +30,7 @@
       expect(data).to have_key(:date)
 
       expect(data[:userId]).to eq(user.id.to_s)
-      expect(data[:source]).to eq("Paycheck")
+      expect(data[:source]).to eq("paycheck")
```
Comment (shawndcarpenter) [inline spec/graphql/mutations/incomes/create_income_request_spec.rb:36]: This all looks well tested in the graphql mutation specs. I can see that it's all working, but I'm wondering if we should add model testing for some of the methods

**Review by trpubz — APPROVED** [review]
working on local

**Pull request merged**